src/main/java/com/github/nkzawa/engineio/client/Socket.java

```diff
@@ -305,6 +305,8 @@ private void probe(final String name) {
         final Transport[] transport = new Transport[] {this.createTransport(name)};
         final boolean[] failed = new boolean[] {false};
         final Socket self = this;
+        
+        self.emit(EVENT_TRANSPORT, transport);
```

Comment:
I'd like to pass the `Transport` instance rather than `Transport` array.
And Maybe we can move this line and others to `createTransport` method?